extensions/wikia/PortabilityDashboard/PortabilityDashboardModel.class.php

```diff
@@ -5,38 +5,93 @@ class PortabilityDashboardModel {
 	const WIKI_ID_FIELD = 'wiki_id';
 	const CUSTOM_INFOBOXES_FIELD = 'custom_infoboxes';
 	const TYPELESS_FIELD = 'typeless';
+	const WIKIS_LIMIT = 500;
 
 	private $connection;
 
-	const WIKIS_LIMIT = 500;
-
 	public function __construct( $db = null ) {
 		$this->connection = $db;
 	}
 
 	/**
-	 * gets model data for portability dashboard
+	 * gets full (non filtered) list of data for portability dashboard
 	 * @return array
 	 */
 	public function getList() {
 		$rowList = $this->getRowList();
+
+		return $this->extendList( $rowList );
+	}
+
+	/**
+	 * For wiki url or domain name return portability data
+	 *
+	 * @param $wikiUrl
+	 * @return array
+	 */
+	public function getWikiByUrl( $wikiUrl ) {
+		$wiki = [ ];
+		$wikiId = WikiFactory::UrlToID( $wikiUrl );
+
+		if ( $wikiId ) {
+			$dataRow = $this->getWikiById( $wikiId );
+
+			if ( $dataRow ) {
+				$wiki = $this->extendList( $dataRow );
```

Comment:
As we discussed offline, maybe this extension would be better done in sequence after the filtering